src/ORM/Table.php

```diff
@@ -1848,6 +1866,24 @@ public function delete(EntityInterface $entity, $options = [])
     }
 
     /**
+     * Try to delete an entity and fail if delete was not successful
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to remove.
+     * @param array|\ArrayAccess $options The options for the delete.
+     * @return bool success
+     * @throws \RuntimeException When the entity couldn't be deleted
+     */
+    public function deleteOrFail(EntityInterface $entity, $options = [])
+    {
+        $deleted = $this->delete($entity, $options);
+        if ($deleted === false) {
+            throw new RuntimeException(sprintf('Could not delete %s with id: %s', $entity->source(), $entity->id));
```

Comment:
Could we generalize the delete failed/save failed exceptions into a `PersistenceFailedException` or something along those lines?